gammapy/visualization/datasets.py

```diff
@@ -0,0 +1,175 @@
+import matplotlib.pyplot as plt
+
+__all__ = [
+    "plot_spectrum_datasets_off_regions",
+    "plot_npred_signal",
+]
+
+
+def plot_spectrum_datasets_off_regions(
+    datasets, ax=None, legend=None, legend_kwargs=None, **kwargs
+):
+    """Plot the off regions of spectrum datasets.
+
+    Parameters
+    ----------
+    datasets : `~gammapy.datasets.Datasets` or list of `~gammapy.datasets.SpectrumDatasetOnOff`
+        List of spectrum on-off datasets.
+    ax : `~astropy.visualization.wcsaxes.WCSAxes`
+        Axes object to plot on.
+    legend : bool
+        Whether to add/display the labels of the off regions in a legend. By default True if
+        ``len(datasets) <= 10``.
+    legend_kwargs : dict
+        Keyword arguments used in `matplotlib.axes.Axes.legend`. The ``handler_map`` cannot be
+        overridden.
+    **kwargs : dict
+        Keyword arguments used in `gammapy.maps.RegionNDMap.plot_region`. Can contain a
+        `~cycler.Cycler` in a ``prop_cycle`` argument.
+
+    Notes
+    -----
+    Properties from the ``prop_cycle`` have maximum priority, except ``color``,
+    ``edgecolor``/``color`` is selected from the sources below in this order:
+    ``kwargs["edgecolor"]``, ``kwargs["prop_cycle"]``, ``matplotlib.rcParams["axes.prop_cycle"]``
+    ``matplotlib.rcParams["patch.edgecolor"]``, ``matplotlib.rcParams["patch.facecolor"]``
+    is never used.
+
+    Examples
+    --------
+    Plot forcibly without legend and with thick circles::
+
+        plot_spectrum_datasets_off_regions(datasets, ax, legend=False, linewidth=2.5)
+
+    Plot that quantifies the overlap of off regions::
+
+        plot_spectrum_datasets_off_regions(datasets, ax, alpha=0.3, facecolor='black')
+
+    Plot that cycles through colors (``edgecolor``) and line styles together::
+
+        plot_spectrum_datasets_off_regions(datasets, ax, prop_cycle=plt.cycler(color=list('rgb'), ls=['--', '-', ':']))  # noqa: E501
+
+    Plot that uses a modified `~matplotlib.rcParams`, has two legend columns, static and
+    dynamic colors, but only shows labels for ``datasets1`` and ``datasets2``. Note that
+    ``legend_kwargs`` only applies if it's given in the last function call with ``legend=True``::
+
+        plt.rc('legend', columnspacing=1, fontsize=9)
```

Comment:
Example code lines have to be preceded by `>>>` for doc tests I think.
See https://github.com/gammapy/gammapy/blob/5fbb60ba48ab9a04143b1b451b1085a4193f8d12/gammapy/estimators/map/excess.py#L108

Reply:
The examples should be fully executable, no.